### PR TITLE
[wip]  BublePlugin

### PR DESCRIPTION
**Pull request by aretecode**
🔋🔌

### CODE REVIEW

File: src/index.optimized.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    configurable: true,
+    enumerable: true,
+    get() {
+      return require(path)
+    },
+  })
+}
+
+// set our exports
+const exporting = {
```
Comment (nchanged) [inline src/index.optimized.ts]: Are you absolutely sure about these dynamic exports?

Comment (nchanged) [inline src/index.optimized.ts]: Oh i see, you let me dwell on in a different file ;-)

**Comment by nchanged** [comment]
Lol James, 

```
src/plugins/js-transpilers/BublePlugin.ts(1,13): error TS6133: 'fs' is declared but never used.
```

File: src/index.optimized.ts (hunk 1 above)
Comment (aretecode) [inline src/index.optimized.ts]: yup :-] I put in another file, would be cool to show the memory perf diff

**Comment by nchanged** [comment]
Still failing ;-( please run `gulp dist`

File: src/analysis/plugins/OwnBundle.ts [hunk 2]
```diff
@@ -36,16 +36,19 @@ export class OwnBundle {
...
             file.analysis.dependencies = [];
             file.isFuseBoxBundle = true;
             // No need in extra footers
             this.removeFuseBoxApiFromBundle(file);
             const externalCollection = file.collection.name !== file.context.defaultPackageName;
+            
             if (externalCollection) {
                 // Ignore this collection as it will be override by the actual bundle
                 file.collection.acceptFiles = false;
+                console.log("here..", file.collection.name);
```
Comment (aretecode) [inline src/analysis/plugins/OwnBundle.ts]: note

File: src/Utils.ts [hunk 3]
```diff
@@ -148,6 +155,16 @@ export function hashString(text: string) {
     return data.toString(16);
 }
 
+export function fastHash(text: string) {
```
Comment (aretecode) [inline src/Utils.ts]: link

File: modules/fuse-box-responsive-api/index.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
+(function() {
+    /* @if universal */
+    var isBrowser = typeof window !== "undefined";
```
Comment (nchanged) [inline modules/fuse-box-responsive-api/index.js]: what is this file doing in master?

File: src/Utils.ts (hunk 3 above)
Comment (nchanged) [inline src/Utils.ts]: what link?

File: src/bundle-abstraction/BundleAbstraction.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { ProducerAbstraction } from "./ProducerAbstraction";
```
Comment (nchanged) [inline src/bundle-abstraction/BundleAbstraction.ts]: what is it all doing in master?

**Pull request merged**